src/memory.cpp

```diff
@@ -76,44 +100,50 @@ std::pair<expr, expr> SingleArrayMemory::load(
 }
 
 
-MultipleArrayMemory::MultipleArrayMemory(unsigned int numBlocks):
-    Memory(ulog2(numBlocks), numBlocks) {
-  for (unsigned i = 0; i < numBlocks; ++i) {
+MultipleArrayMemory::MultipleArrayMemory(
+    unsigned int numGlobalBlocks, unsigned int nunLocalBlocks):
+  Memory(numGlobalBlocks, nunLocalBlocks, ulog2(numGlobalBlocks + nunLocalBlocks)) {
+  for (unsigned i = 0; i < getNumBlocks(); ++i) {
     auto suffix = [&](const string &s) {
       return s + to_string(i);
     };
     arrays.push_back(ctx.constant(suffix("array").c_str(),
         ctx.array_sort(Index::sort(), Float::sort())));
     writables.push_back(ctx.bool_const(suffix("writable").c_str()));
     numelems.push_back(ctx.bv_const(suffix("numelems").c_str(), Index::BITS));
+    isGlobals.push_back(ctx.bool_val(i < numGlobalBlocks));
```

Comment:
BTW, let's leave this scheme as a comment at somewhere:
```
// Bids smaller than numGlobalBlocks are global
```

Reply:
I added it https://github.com/aqjune/mlir-tv/pull/79/files#diff-3d8d4950551437b1a628fd00613869df6939a434606baa3cf36bd391aa463154R49-R52.